tsdb/compact_test.go

```diff
@@ -1298,29 +1299,79 @@ func TestDeleteCompactionBlockAfterFailedReload(t *testing.T) {
 
 func TestHeadCompactionWithHistograms(t *testing.T) {
 	head, _ := newTestHead(t, DefaultBlockDuration, false)
+	require.NoError(t, head.Init(0))
 	t.Cleanup(func() {
 		require.NoError(t, head.Close())
 	})
 
-	require.NoError(t, head.Init(0))
-	app := head.Appender(context.Background())
+	minute := func(m int) int64 { return int64(m) * time.Minute.Milliseconds() }
+	ctx := context.Background()
+	appendHistogram := func(lbls labels.Labels, from, to int, h *histogram.Histogram, exp *[]tsdbutil.Sample) {
+		t.Helper()
+		app := head.Appender(ctx)
+
+		for tsMinute := from; tsMinute <= to; tsMinute++ {
+			_, err := app.AppendHistogram(0, lbls, minute(tsMinute), h)
+			require.NoError(t, err)
+			*exp = append(*exp, sample{t: minute(tsMinute), h: h.Copy()})
+		}
 
-	type timedHistogram struct {
-		t int64
-		h *histogram.Histogram
+		require.NoError(t, app.Commit())
+	}
+	appendFloat := func(lbls labels.Labels, from, to int, exp *[]tsdbutil.Sample) {
+		t.Helper()
+		app := head.Appender(ctx)
+		for tsMinute := from; tsMinute <= to; tsMinute++ {
+			_, err := app.Append(0, lbls, minute(tsMinute), float64(tsMinute))
+			require.NoError(t, err)
+			*exp = append(*exp, sample{t: minute(tsMinute), v: float64(tsMinute)})
+		}
+		require.NoError(t, app.Commit())
 	}
 
-	// Ingest samples.
-	numHistograms := 120 * 4
-	timeStep := DefaultBlockDuration / int64(numHistograms)
-	expHists := make([]timedHistogram, 0, numHistograms)
-	l := labels.Labels{{Name: "a", Value: "b"}}
-	for i, h := range GenerateTestHistograms(numHistograms) {
-		_, err := app.AppendHistogram(0, l, int64(i)*timeStep, h)
-		require.NoError(t, err)
-		expHists = append(expHists, timedHistogram{int64(i) * timeStep, h})
+	// Append all scenario of histograms.
```

Comment:
I'm confused by this doc comment. Maybe "Append floats and histograms for all scenarios."? Not sure what you want to express. Maybe no comment is needed?